### PR TITLE
Feat/hng 126 help center page

### LINKED ISSUE #126: [Feature]: Create Help Center Page
### Description

Create and Implement the Help Center Page. This page must be fully responsive and implemented according to the design

### Acceptance Criteria

- Page must be fully responsive on mobile, tablets and desktops

- Page must fully implement what was designed

- Page must use reusable components already existing

### Test

- Test page is fully responsive
- Ensure page uses existing components
- Ensure buttons and links works appropriately

### Links

https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=4-21184&m=dev

### Image
Screenshot 2024-07-18 at 08 21 08

### PR DESCRIPTION
**Pull request by frank1003A**
# Description

**Closes #126**

# Changes proposed

## What were you told to do?

Implement the Help Center page as per the specifications provided in the issue #126.

## What did you do?

- Created a new Help Center page component.
- Added routing to the Help Center page.
- Implemented the Topics accordion component for the Help Center.
- Styled the Help Center page to match the design specifications.
- Added unit tests for the FAQ accordion component to ensure functionality.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos

ts5
localhost_3001_help-center

### CODE REVIEW

File: src/components/common/FaqAccordion/FaqAcccordion.test.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      "Yes, the application is accessible from multiple devices including desktops, tablets, and smartphones. Your data will be synced across all devices, ensuring a seamless experience.",
+  },
+];
+
+describe("faqAccordion", () => {
+  it("renders correctly with given FAQ data", () => {
+    expect.hasAssertions();
+    render(<FaqAccordion faqs={faqs} />);
+
+    // Check if the questions are rendered
```
Comment (taiwonaf) [inline src/components/common/FaqAccordion/FaqAcccordion.test.tsx]: Remove

File: src/components/common/FaqAccordion/FaqAcccordion.test.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  it("renders correctly for different screen sizes", () => {
+    expect.hasAssertions();
+    render(
+      <FaqAccordion
+        faqs={faqs}
+        containerClassName="p-6 bg-neutral-50 w-full md:w-96 rounded-xl"
+      />,
+    );
+
+    // Check if the container class includes responsive width
```
Comment (taiwonaf) [inline src/components/common/FaqAccordion/FaqAcccordion.test.tsx]: Remove

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,9 +1,11 @@
 {
```
Comment (taiwonaf) [inline .vscode/settings.json]: What folder is this ?

File: src/components/common/FaqAccordion/FaqAcccordion.test.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/* eslint-disable prettier/prettier */
```
Comment (taiwonaf) [inline src/components/common/FaqAccordion/FaqAcccordion.test.tsx]: Fix this locally

File: src/components/layouts/Helpcenterpage/__snapshots__/Helpcenterpage.test.tsx.snap [hunk 5]
```diff
@@ -0,0 +1,2353 @@
+// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html
```
Comment (taiwonaf) [inline src/components/layouts/Helpcenterpage/__snapshots__/Helpcenterpage.test.tsx.snap:1]: What is this ?

File: src/components/layouts/Helpcenterpage/index.tsx [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+const handleButtonClickTest = () => {
+  alert("Contact Button Click Test");
+};
+
+const HelpCenterPage = () => {
+  return (
+    <div className="w-full bg-background">
+      <div className="w-full bg-primary/10">
+        <div className="mx-auto w-full" style={{ maxWidth: "1349px" }}>
```
Comment (taiwonaf) [inline src/components/layouts/Helpcenterpage/index.tsx]: Why not tailwind class?

File: src/components/layouts/Helpcenterpage/index.tsx [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+                aria-level={1}
+              >
+                How can we help You?
+              </h1>
+              <p className="text-center text-base font-normal text-neutral-600 md:text-lg">
+                Find advice and answers from our support team
+              </p>
+              <div className="flex h-fit w-full items-center justify-start overflow-hidden rounded-xl border border-slate-300 bg-white px-2 py-[2px] text-xs font-normal leading-none text-neutral-600 focus-visible:ring-0 focus-visible:ring-offset-0 md:w-[466px] md:rounded-3xl">
+                <span className="flex h-8 w-8 items-center justify-center p-1">
+                  <svg
```
Comment (taiwonaf) [inline src/components/layouts/Helpcenterpage/index.tsx]: Don't use directly in code, check Lucide for it or create a separate file for it

File: src/components/layouts/Helpcenterpage/index.tsx [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+                  >
+                    <path
+                      d="M20 20L16.2223 16.2156M18.3158 11.1579C18.3158 13.0563 17.5617 14.8769 16.2193 16.2193C14.8769 17.5617 13.0563 18.3158 11.1579 18.3158C9.2595 18.3158 7.43886 17.5617 6.0965 16.2193C4.75413 14.8769 4 13.0563 4 11.1579C4 9.2595 4.75413 7.43886 6.0965 6.0965C7.43886 4.75413 9.2595 4 11.1579 4C13.0563 4 14.8769 4.75413 16.2193 6.0965C17.5617 7.43886 18.3158 9.2595 18.3158 11.1579V11.1579Z"
+                      stroke="#525252"
+                      strokeWidth="1.5"
+                      strokeLinecap="round"
+                    />
+                  </svg>
+                </span>
+                <Input
```
Comment (taiwonaf) [inline src/components/layouts/Helpcenterpage/index.tsx:43]: Use custom input

File: src/components/layouts/Helpcenterpage/index.tsx [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+                  placeholder="Search on any topic..."
+                  className="w-full border-none bg-transparent px-1 py-2 focus-visible:outline-none focus-visible:ring-0 focus-visible:ring-offset-0"
+                  aria-label="Search on any topic"
+                />
+              </div>
+            </div>
+          </section>
+        </div>
+      </div>
+      <div className="mx-auto w-full" style={{ maxWidth: "1349px" }}>
```
Comment (taiwonaf) [inline src/components/layouts/Helpcenterpage/index.tsx]: Use tailwind class

File: src/components/layouts/Helpcenterpage/index.tsx [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+          className="flex flex-col items-center justify-start gap-7 px-6 py-24 md:px-24"
+          aria-labelledby="browse-topics-heading"
+        >
+          <span
+            id="browse-topics-heading"
+            className="w-96 text-center text-3xl font-bold text-orange-500"
+          >
+            Browse by topics
+          </span>
+          {/** Tooltip Accordions */}
```
Comment (taiwonaf) [inline src/components/layouts/Helpcenterpage/index.tsx]: Remove comment

**Review by taiwonaf — CHANGES_REQUESTED** [review]

File: src/components/layouts/Helpcenterpage/__snapshots__/Helpcenterpage.test.tsx.snap (hunk 5 above)
Comment (frank1003A) [reply src/components/layouts/Helpcenterpage/__snapshots__/Helpcenterpage.test.tsx.snap:1]: Generated snapshots by test for responsiveness tests

File: src/components/common/FaqAccordion/FaqAcccordion.test.tsx (hunk 2 above)
Comment (frank1003A) [reply src/components/common/FaqAccordion/FaqAcccordion.test.tsx]: done

File: src/components/layouts/Helpcenterpage/index.tsx (hunk 6 above)
Comment (frank1003A) [reply src/components/layouts/Helpcenterpage/index.tsx]: Modified

File: src/components/layouts/Helpcenterpage/index.tsx (hunk 7 above)
Comment (frank1003A) [reply src/components/layouts/Helpcenterpage/index.tsx]: Resolved

File: src/components/layouts/Helpcenterpage/index.tsx (hunk 9 above)
Comment (frank1003A) [reply src/components/layouts/Helpcenterpage/index.tsx]: Resolved

File: src/components/layouts/Helpcenterpage/index.tsx (hunk 10 above)
Comment (frank1003A) [reply src/components/layouts/Helpcenterpage/index.tsx]: resolved

File: src/components/layouts/Helpcenterpage/index.tsx (hunk 8 above)
Comment (frank1003A) [reply src/components/layouts/Helpcenterpage/index.tsx:43]: resolved

**Pull request closed without merging**